### ISSUE TITLE
Journal-wide search facilities

### ISSUE THREAD

**Issue by vkazanov** [bastibe/org-journal#27]
At some point I realized that basic calendar intergration is not enough for the journal - manually searching dozens of files just does not scale very well. Agenda-based search does not scale to dozens of files too.

One option is to use generic Emacs tools, but this does not allow for smart queries such as "journal entries with the tag 'tag' during the previous week".

So, I am working on specialized search system for the org-journal. Currently my code is mostly tag-based (queries like "find entries with the following tags" are possible).

I am considering other, more powerful, means of searching (such as "find entries with a given text over a given period").

The question is: what do you, the creator of the package, use for searching and categorizing journal entries? This would help to adapt my code to your needs too.

**Comment by bastibe** [comment]
That's a very interesting idea! I would *love* to be able to search a certain time-slice in the journal! Having a keyboard shortcut in the calendar that searches the current week or month would be genius!

At the moment, I am mostly using agenda searches, and relatively quickly-changing tags. For longer projects, I tend to move my development notes to a separate org file because agenda searches just don't work any more. But agenda searches really don't scale very well, and always open *all* journal files. It's not a perfect solution.

**Comment by vkazanov** [comment]
I use the journal to write notes about things I did, movies I watched, code I wrote, daytime work notes, etc. Sometimes it is nice to get, say, a list of movie-related journal entries I watched last week... 

I guess that plain text search is okay here, as for tag searches I can just search for something like ":movie:".

So... How about the following: I will add a function doing a period-based search, with a few calendar-specific bindings doing month/week/etc searches? The interface will probably be a bit spartan, as I couldn't make up my mind about a proper way to display search results, it's just insert-text-button for now.
